cartographer_ros/src/cartographer_node_main.cc

```diff
@@ -90,6 +93,18 @@ Rigid3d ToRigid3d(const geometry_msgs::TransformStamped& transform) {
                                     transform.transform.rotation.z));
 }
 
+Rigid3d ToRigid3d(const geometry_msgs::Pose& pose) {
+  return Rigid3d(
+      Eigen::Vector3d(pose.position.x, pose.position.y, pose.position.z),
+      Eigen::Quaterniond(pose.orientation.w, pose.orientation.x,
+                         pose.orientation.y, pose.orientation.z));
+}
+
+PoseCovariance ToPoseCovariance(boost::array<double, 36ul> covariance) {
```

Comment:
I am unsure about the semantics here - but it seems covariance is copied and Eigen::Map<> then references this temporary - since it does not own data. This can be fixed by taking covariance as a const ref&, but it still requires a comment that the returned value must not outlive the passed in value. 

Maybe I got it all wrong though.